src/pages/history/components/collapse-content/collapse-content.tsx

```diff
@@ -1,47 +1,133 @@
-import Box from '@mui/material/Box';
-import Typography from '@mui/material/Typography'
+import AceEditor from "react-ace/";
+import "ace-builds/src-noconflict/mode-json";
+import "ace-builds/src-noconflict/theme-clouds_midnight";
+import "ace-builds/src-noconflict/theme-solarized_dark";
+import "ace-builds/src-noconflict/theme-dracula";
 
-import _isEmpty from 'lodash/isEmpty'
+import _isEmpty from "lodash/isEmpty";
 
-import { TState } from 'models/state'
+import { TState } from "models/state";
 
-import * as S from './styles'
+import * as S from "./styles";
+import ContentCopyOutlinedIcon from "@mui/icons-material/ContentCopyOutlined";
+import { IconButton } from "shared/components/icon-button";
 
 type Props = {
   state: TState;
-}
+};
 
 export const CollapseContent: React.FC<Props> = ({ state }) => {
+  function onClickCopyBag() {
+    navigator.clipboard.writeText(JSON.stringify(state.bag, null, "\t"));
+  }
+
+  function onClickCopyResult() {
+    navigator.clipboard.writeText(JSON.stringify(state.result, null, "\t"));
+  }
+
+  function onClickCopyActor() {
```

Comment:
Essas 3 funções para copiar podem ser escritas como uma função e ser passada via _props_ para os components que usam essa função.

```suggestion
  function copyToClipboard(value: string) {
     navigator.clipboard.writeText(JSON.stringify(value, null, "\t"));
  }
```